### PR TITLE
Add tweet functionality

### PR DESCRIPTION
**Pull request by fusillicode**
Together with the tweet functionality I've also refactored the call to show the first quote when the page gets loaded.

Obviously, if there is something wrong don't hesitate to contact me ;)

### CODE REVIEW

File: tweet.js [hunk 1]
```diff
@@ -0,0 +1,46 @@
...
+function prepareMessage(popUp) {
+  var tweetLengthLimit = 140;
+  var tweetUrlLengthLimitForHTTP = 22;
+  var tweetUrlLengthLimitForHTTPS = 23;
+  var trimString = '... ';
+  var tweetUrlLengthLimit = window.location.protocol === "https:" ?
+                            tweetUrlLengthLimitForHTTPS :
+                            tweetUrlLengthLimitForHTTP;
+  var tweetTextLengthLimit = tweetLengthLimit - tweetUrlLengthLimit - trimString.length;
+  return escape(popUp.message).substring(0, tweetTextLengthLimit) + (tweetTextLengthLimit < popUp.message.length ? trimString : ' ') + popUp.url;
```
Comment (nocturnalgeek) [inline tweet.js]: I think popUp.url is not required as it automatically adds it. Right now, the url is coming twice in the tweet!

Comment (fusillicode) [inline tweet.js]: Locally I don't get automatically the url but anyway you can safely remove it from the message that gets constructed if it's actually duplicated :)

Comment (nocturnalgeek) [inline tweet.js]: See here http://fusillicode.github.io/ChuckNorrisGeekyFacts/

Comment (fusillicode) [inline tweet.js]: Checked and now should be fixed :)

File: tweet.js [hunk 2]
```diff
@@ -0,0 +1,47 @@
...
+  var popUp = $.extend(defaultAttributes, customAttributes);
+  popUp.message = prepareMessage(popUp);
+  return popUp;
+}
+
+function prepareMessage(popUp) {
+  var tweetLengthLimit = 140;
+  var tweetUrlLengthLimitForHTTP = 22;
+  var tweetUrlLengthLimitForHTTPS = 23;
+  var trimString = '... ';
```
Comment (nocturnalgeek) [inline tweet.js]: The url is automatically appended with a space before it. Remove the space after the dots. Right now, there are two spaces before the url.

File: tweet.js [hunk 3]
```diff
@@ -0,0 +1,47 @@
...
+  var tweetUrlLengthLimit = window.location.protocol === "https:" ?
+                            tweetUrlLengthLimitForHTTPS :
+                            tweetUrlLengthLimitForHTTP;
+  var tweetTextLengthLimit = tweetLengthLimit - tweetUrlLengthLimit - trimString.length;
+  var message = popUp.message.substring(0, tweetTextLengthLimit) + (tweetTextLengthLimit < popUp.message.length ? trimString : ' ');
```
Comment (nocturnalgeek) [inline tweet.js]: Same as above, remove the extra space.

Comment (fusillicode) [inline tweet.js]: Ok I think I've done that too :)

**Comment by nocturnalgeek** [comment]
Just one more thing, if the message is long, in that case the total length of the tweet is coming out to be 141!

**Comment by fusillicode** [comment]
Fixed, rapidly but fixed ;)

P.S: anyway the code could be really optimized and refactored ;)

**Comment by nocturnalgeek** [comment]
Yes :)
Anyways, its working as expected.
Thanks @fusillicode! :+1:

**Pull request merged**